api/src/main/java/bio/terra/tanagra/service/search/NameUtils.java

```diff
@@ -0,0 +1,33 @@
+package bio.terra.tanagra.service.search;
+
+import com.google.common.base.Preconditions;
+import java.util.regex.Pattern;
+
+/**
+ * Utilities for checking Tanagra names.
+ *
+ * <p>Some names traverse Tanagra from HTTP requests to SQL generation. For these names, it's
+ * important that they are relatively friendly and simple.
+ */
+final class NameUtils {
```

Comment:
Should this be applied to variables?

Reply:
Maybe. Variables are only user specified and are used directly in SQL, so they're a little differnt. OTOH, having a consistent simple naming schema might be valuable. I could go either way, WDYT?

Reply:
I can definitely see them needing to diverge for that reason (or else necessitating a conversion layer for Tanagra variable -> SQL variable). I'd keep them semantically separate, e.g. in API documentation - but your call whether you want to just reuse this code, or split out a different utility.

Reply:
reverting variable to keep it separate for now.